R/explore.R

```diff
@@ -121,6 +121,220 @@ mc_plot_univariate <- function(i, x, dist="ZINB") {
+                                   colour = "black")
+      }
+
+    } else {
+      d <- data.frame(Surveyed = srv, Variable = z)
+
+      p <- ggplot2::ggplot(
+        data.frame(
+          Surveyed = srv,
+          Variable = z),
+        ggplot2::aes(
+          x = .data$Variable,
+          fill = .data$Surveyed,
+          color = .data$Surveyed)) +
+        ggplot2::geom_rug(data = data.frame(
+          Surveyed = srv,
+          Variable = z)[!srv,], sides="b") +
+        ggplot2::geom_rug(data = data.frame(
+          Surveyed = srv,
+          Variable = z)[srv,], sides="t") +
+        ggplot2::theme_minimal() +
+        ggplot2::scale_fill_viridis_d(end = 0.8, aesthetics = c("fill", "colour")) +
+        ggplot2::ylab("Density") +
+        ggplot2::xlab(i)
+
+      if(interactive) {
+        d <- dplyr::mutate(d, tt = paste0("Surveyed: ", .data$Surveyed))
+        p <- p + ggiraph::geom_density_interactive(
+          data = d, alpha = 0.50,
+          ggplot2::aes(tooltip = .data$tt, data_id = .data$tt))
+      } else {
+       p <- p + ggplot2::geom_density(alpha = 0.5)
+      }
+
+    }
+  }
+  if (type == "fit") {
+    if (CAT) {
+      p <- ggplot2::ggplot(
+        dat,
+        ggplot2::aes(
+          y = .data$y,
+          x = .data$z)) +
+        ggplot2::theme_minimal() +
+        ggplot2::ylab("Total Moose") +
+        ggplot2::xlab(i)
+
+      tot <- stats::aggregate(dat[, c("y", "zhat")], list(z = dat$z), mean)
+      tot$z_int <- as.integer(tot$z)
+
+      if(interactive) {
+        out <- dplyr::group_by(dat, .data$z)
+        out <- dplyr::mutate(out,
+                             IQR = stats::IQR(.data$y),
+                             q75 = stats::quantile(.data$y, 0.75),
+                             q25 = stats::quantile(.data$y, 0.25))
+        out <- dplyr::filter(out,
+                             .data$y > (.data$q75 + (1.5 * .data$IQR)) |
+                               .data$y < (.data$q25 - (1.5 * .data$IQR)))
+        out <- dplyr::mutate(out, tt = paste0("SU_ID: ", .data$SU_ID,
+                                              "<br>", .env$i, ": ", .data$z,
+                                              "<br>Total Moose: ", .data$y))
+        dat <- dplyr::mutate(dat, tt = .data$z)
+        tot <- dplyr::mutate(tot, tt = paste0(.data$z,
+                                              "<br>Mean: ", round(.data$zhat, 2)))
+
+        p <- p +
+          ggiraph::geom_boxplot_interactive(
+            data = dat, fill = "grey", outlier.shape = NA, # omit outliers
+            ggplot2::aes(tooltip = .data$tt, data_id = .data$tt)) +
+          ggiraph::geom_point_interactive(                 # manually add outliers
+            data = out, size = 3, alpha = 0.75,
+            ggplot2::aes(tooltip = .data$tt, data_id = .data$tt)) +
+          ggiraph::geom_point_interactive(
+            data = tot, size = 3, col = 4,
+            ggplot2::aes(tooltip = .data$tt, data_id = .data$tt))
```

Comment:
To make the boxplot outlier point tooltips useful, I had to calculate outliers by hand and plot them separately with geom_point_interactive().

Reply:
I love these plots. So fancy!